### ISSUE TITLE
Error when runing apt-select

### ISSUE THREAD

**Issue by tiberiuichim** [jblakeman/apt-select#11]
```
Getting list of mirrors ... done.
Testing 6 mirror(s) [5/6] 83%Could not resolve hostname
[Errno -3] Temporary failure in name resolution
Traceback (most recent call last):
  File "apt-select.py", line 177, in <module>
    lowest = ping.minRTT()
  File "/home/tibi/work/apt-select/mirrors.py", line 49, in minRTT
    rtt = self.__tcpPing()
  File "/home/tibi/work/apt-select/mirrors.py", line 36, in __tcpPing
    s.connect((self.addr, port))
AttributeError: RoundTrip instance has no attribute 'addr'
```

**Comment by jblakeman** [comment]
A check has been added in [da8e07a](https://github.com/jblakeman/apt-select/commit/da8e07a8863da74ab27d6565f3dfb639c94f530f) to exit on name resolution failures.

**Comment by tiberiuichim** [comment]
the solution is wrong. The traceback says that it's at 83% checking, it resolved 5 out of 6 hosts. Obviously the net connection was working and the fix to just exit is not a good one.

**Comment by jblakeman** [comment]
It looks like this error can be caused by many things including [improper DNS forwarding](https://askubuntu.com/questions/91543/apt-get-update-fails-to-fetch-files-temporary-failure-resolving-error) or a [misconfigured DHCP server on your network](https://serverfault.com/questions/266042/temporary-failure-in-name-resolution/329009#329009).  Depending on the cause, the problem probably won't just go away.  If you run the script with these DNS errors, and the list of mirrors is long, it would needlessly try to resolve each mirror name before reporting nothing, and there would likely be later [problems](https://answers.launchpad.net/ubuntu/+source/apt/+question/197664) updating or installing using `apt-get`.

That said, I agree that error handling shouldn't ignore unlikely scenarios, so exiting may not be the best solution for this or something like an outdated or dead mirror address even though it's not likely that Canonical would provide a bad address in `mirrors.txt`.

**Comment by jblakeman** [comment]
Addressed in [467917c](https://github.com/jblakeman/apt-select/commit/467917c589ab9821bec7d129b6e560f3ff89bfee)
